script.js

```diff
@@ -1,121 +1,145 @@
-var input = document.querySelectorAll('input');
+var myLibrary = [];
 
-let newBook = document.querySelector('#add-new-book');
-let modal = document.querySelector('#modal');
+let library = document.getElementById('container');
+let addBookButton = document.getElementById('add-new-book');
+let form = document.getElementById('new-book');
+let modal = document.getElementById('modal');
 let overlay = document.querySelector('.overlay');
-let submitBook = document.querySelector('#submit-book');
-let container = document.querySelector('.container');
-let form = document.querySelector('form');
 
+addBookButton.addEventListener('click', toggleModal);
+overlay.addEventListener('click', toggleModal);
 
-newBook.addEventListener('click', addNewBook);
-overlay.addEventListener('click', closeModal);
-submitBook.addEventListener('click', bookDetails);
+const addBookToLibrary = (e) => {
+  e.preventDefault();
+  let bookData = [];
+  let input = document.querySelectorAll('input');
 
-function addNewBook() {
-    modal.classList.remove('disappear');
-    modal.classList.add('appear');
-    newBook.classList.remove('antirotate');
-    newBook.classList.add('rotate');
-    overlay.classList.add('active');
-}
+  for (let i = 0; i < input.length; i++) {
+    if (input[i].type === "checkbox") {
+      bookData.push(input[i].checked);
+    } else {
+      bookData.push(input[i].value);
+    }
+  }
 
-function closeModal() {
-    modal.classList.remove('appear');
-    modal.classList.add('disappear');
-    overlay.classList.remove('active');
-    newBook.classList.remove('rotate');
-    newBook.classList.add('antirotate');
-    
-    form.reset();
+  const newBook = new Book(...bookData);
+  myLibrary.push(newBook.bookCardContent());
+  displayLibrary();
+  toggleModal();
 }
 
-function bookDetails() {
-    let bookTitle = document.querySelector('[name="title"]');
-    let bookAuthor = document.querySelector('[name="author"]');
-    let bookYear = document.querySelector('[name="year"]');
-    let bookPages = document.querySelector('[name="pages"]');
-    let bookRead = document.querySelector('[name="read"]');
-    
-    let book = new Book(bookTitle.value, bookAuthor.value, bookYear.value, bookPages.value, bookRead.checked);
-    
-    if (form.checkValidity()) createBook(book);
-}
-
-function createBook(book) {
+function displayLibrary() {
+  form.reset();
```

Comment:
This function only purpose should be to display the library items, resetting a form is a side effect it should not be concerned about. Move this to form event listener.